devops/azure-pipelines/discord-job-template.yml

```diff
@@ -5,26 +5,26 @@ parameters:
     type: string
   - name: 'title'
     type: string
-  - name: 'discordVarGroup'
+  - name: 'discordVarGroup' # Secret discord information, channelID and key
```

Comment:
not descriptive enough. i can't tell from looking at this line that i need two variables, `discordid` which is the channel ID, and `discordkey` which is the key (i guess that's unambiguous enough in context?).

Reply:
We feel it is descriptive enough in context of discord webhooks. What would you recommend we change it to make it more clear?

Reply:
Would it be good enough if we make sure there's documentation like "we created a discord variable group with this name, and these variables in the group" ?
because this name is a reference to a specific variable group defined outside the pipeline, so that we're not hard-coding the name of the variable group in the discord job template. (though i'm realizing we've just hard-coded the names of the child variables of the variable group). 
It's feeling like a tradeoff between parameterization and minimizing duplication.

Reply:
I'll try to reorganize the variables, make the var group global, and pass the child vars as parameters into the template. That should avoid any duplication, and be also pretty transparent.

Reply:
My attempt failed seemingly due to variables and parameters needing to be available at different points in time. Not sure if I can work around it yet.

Reply:
the way i think it works is template & parameter substitution happens at 'compile time', so it might replace the string `${{ parameters.data }}` with the string `$(myData)`, which at runtime will be interpreted as a variable reference.
anyway, what i'm requiring with this review comment is simply to document the fact that "if you want this pipeline to work, the variable group named ____ needs two variables, ___ (with the ___) and ___ (with the ___)". imagine someone trying to use this file to set up ci/cd with a different codebase and azure devops project.

Reply:
Yeah, the fact that parameters are needed at compile time, and the variables come later doesn't help when trying to feed a variables into the parameters for the template to use, which, I guess, may be a hint that we can't really pass variables in through parameters in the first place. 

Will add comments promptly.

Reply:
Comments added in-file. Will add documentation in AzDO wiki as well.